pyomo/contrib/matching/tests/test_interface.py

```diff
@@ -0,0 +1,169 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import pyomo.environ as pyo
+from pyomo.common.dependencies import networkx_available
+from pyomo.common.dependencies import scipy_available
+from pyomo.common.collections import ComponentSet
+from pyomo.contrib.matching.interface import IncidenceGraphInterface
```

Comment:
I think you need to trap for `networkx_available` before this import: `interface` imports `block_triangularize` which unconditionally imports `networkx`.  Alternatively, you should remove the unconditional imports (see comment in `block_triangularize` and `maximal_matching`